### PR TITLE
Sparkling Aria shouldn't cure burn if the user faints

### PR DESCRIPTION
**Pull request by pacmanboss256**

### CODE REVIEW

**Comment by Zarel** [comment]
The problem with titles like this is that "no longer" is kind of ambiguous. Was it originally correct, and then it no longer doing so is a bug? Or was it originally bugged, and no longer doing so is the fix?

Generally the ways I find unambiguous are "Fix ... to ..." or "... should ...".

The PR description should also say things like "In Sword/Shield, this happens" and "Previously, in PS, this happens, which is a bug". I constantly have to ask clarifications because things aren't phrased that way.

File: data/moves.ts [hunk 1]
```diff
@@ -16242,10 +16242,16 @@ export const Moves: {[moveid: string]: MoveData} = {
 		secondary: {
 			dustproof: true,
 			chance: 100,
-			onHit(target) {
-				if (target.status === 'brn') target.cureStatus();
+			onHit(target, source, move) {
+				if (target.status === 'brn') {
+					target.addVolatile('cureburn', source, move);
```
Comment (Zarel) [inline data/moves.ts:16247]: Why are you using a volatile to track this? Does Sparkling Aria not cure burns if the target gets burned during the attack?

Comment (pacmanboss256) [reply data/moves.ts:16247]: the target does not get cured if the user of the move faints on the same turn. I needed the volatile so it still is affected by sheer force but not shield dust, but the actual curing is after "onDamage" as in innards out

Comment (Zarel) [reply data/moves.ts:16247]: Oh, it's for Sheer Force.

File: data/moves.ts [hunk 2]
```diff
@@ -16242,10 +16242,16 @@ export const Moves: {[moveid: string]: MoveData} = {
...
+				}
 			},
 		},
+		onAfterMove(source, target, move) {
+			if (!source.fainted && target.getVolatile('cureburn')) target.cureStatus();
+			target.removeVolatile('cureburn');
```
Comment (urkerab) [inline data/moves.ts:16253]:
```suggestion
			if (target.removeVolatile('sparklingaria') && target.status === 'brn' && !source.fainted) target.cureStatus();
```
(see above)

File: test/sim/moves/sparklingaria.js [hunk 3]
```diff
@@ -12,18 +12,17 @@ describe('Sparkling Aria', function () {
 
 	it(`should cure the target's burn`, function () {
 		battle = common.createBattle([[
-			{species: 'Wynaut', ability: 'compoundeyes', moves: ['will-o-wisp', 'sparklingaria']},
+			{species: 'Wynaut', ability: 'compoundeyes', moves: ['will-o-wisp', 'sparklingaria', 'sleeptalk']},
```
Comment (DaWoblefet) [inline test/sim/moves/sparklingaria.js:15]: Don't forget to remove the extra Sleep Talk.

**Review by DaWoblefet — APPROVED** [review]
I think this will end up fixing itself once I finish documenting post-move success checks, but seems fine for now.

**Review by Marty-D — CHANGES_REQUESTED** [review]
Sparkling Aria is a spread move and `AfterMove` only happens once.

**Comment by pacmanboss256** [comment]
bump

**Comment by pacmanboss256** [comment]
Bump

**Comment by Marty-D** [comment]
No need to keep bumping this. I want to check something first and then I'll merge if it's fine.

**Comment by pacmanboss256** [comment]
Gotcha.

**Comment by pacmanboss256** [comment]
its been over 2 weeks, what was it you wanted to check?

File: data/moves.ts [hunk 4]
```diff
@@ -16242,9 +16242,14 @@ export const Moves: {[moveid: string]: MoveData} = {
...
 			chance: 100,
-			onHit(target) {
-				if (target.status === 'brn') target.cureStatus();
-			},
+			volatileStatus: 'sparklingaria',
+		},
+		onAfterMove(source, target, move) {
+			for (const pokemon of this.getAllActive()) {
+				if (pokemon !== source && pokemon.removeVolatile('sparklingaria') && pokemon.status === 'brn' && !source.fainted) {
+					target.cureStatus();
```
Comment (Marty-D) [inline data/moves.ts:16250]: `target` is only one specific Pokemon in an `AfterMove` context; you want `pokemon` here too.

**Review by Marty-D — APPROVED** [review]
Okay this should be fine now, thanks!

**Pull request merged**